File: langchain-4j/src/main/kotlin/org/example/ResultDisplayer.kt
package org.example

/**
 * レシピ抽出結果の表示を担当するクラス（LangChain4j版）
 */
class ResultDisplayer {

    /**
     * レシピ抽出結果を表示する
     */
    fun displayResults(result: RecipeExtractionResult) {
        println("\n✅ === ワークフロー完了！ ===")
        println(buildString {
            appendLine("📊 実行結果:")

            if (result.extractedRecipe != null) {
                appendLine()
                appendLine("📋 RecipeEntityオブジェクト:")
                appendLine("RecipeEntity(")
                appendLine("  name = \"${result.extractedRecipe.name}\",")
                appendLine("  ingredients = listOf(")
                result.extractedRecipe.ingredients.forEach { ingredient ->
                    appendLine("    Ingredient(\"${ingredient.name}\", \"${ingredient.unit}\", ${ingredient.quantity}f),")
                }
                appendLine("  )")
                appendLine(")")

                if (result.totalCookingMinutes != null) {
                    appendLine()
                    appendLine("⌚️ 調理時間: ${result.totalCookingMinutes}分")
                }
            } else {
                appendLine("❌ RecipeEntityの抽出に失敗しました")
            }
        })
    }
}

File: mastra/src/result-displayer.ts
import { RecipeExtractionResult } from './types.js';

/**
 * レシピ抽出結果の表示を担当するクラス
 * KoogとLangChain4jのResultDisplayerに対応するTypeScript版
 */
export class ResultDisplayer {

  /**
   * レシピ抽出結果を表示する
   */
  displayResults(result: RecipeExtractionResult): void {
    console.log("\n✅ === ワークフロー完了！ ===");
    
    let output = "📊 実行結果:\n";

    if (result.extractedRecipe) {
      output += "\n📋 RecipeEntityオブジェクト:\n";
      output += "{\n";
      output += `  name: "${result.extractedRecipe.name}",\n`;
      output += "  ingredients: [\n";
      
      result.extractedRecipe.ingredients.forEach(ingredient => {
        output += `    { name: "${ingredient.name}", unit: "${ingredient.unit}", quantity: ${ingredient.quantity} },\n`;
      });
      
      output += "  ]\n";
      output += "}";

      if (result.totalCookingMinutes !== null) {
        output += `\n\n⌚️ 調理時間: ${result.totalCookingMinutes}分`;
      }
    } else {
      output += "\n❌ RecipeEntityの抽出に失敗しました";
    }

    console.log(output);
  }
}